From 233f808c13df8dbb9425b5dd42cef80dee47e319 Mon Sep 17 00:00:00 2001
From: Don Darling <don.osc2@gmail.com>
Date: Tue, 16 Mar 2010 16:31:51 -0500
Subject: [PATCH 11/12] Add V4L2 ioctl calls to initialize DM6467T and DM365 drivers properly.

After calling the ENUMINPUT on an input, call S_INPUT and G_INPUT on that
input.  Otherwise, the video standards (norms) won't be detected properly.
---
 sys/v4l2/v4l2_calls.c |   15 +++++++++++++++
 1 files changed, 15 insertions(+), 0 deletions(-)

diff --git a/sys/v4l2/v4l2_calls.c b/sys/v4l2/v4l2_calls.c
index 5d486fa..6fb9bfa 100644
--- a/sys/v4l2/v4l2_calls.c
+++ b/sys/v4l2/v4l2_calls.c
@@ -168,6 +168,21 @@ gst_v4l2_fill_lists (GstV4l2Object * v4l2object)
          continue;
     }
 
+    #if !defined(MVL_LSP) && \
+         (defined(Platform_dm6467t) || defined(Platform_dm365))
+    /* Try setting the input and see if we can retrieve it as a valid setting.
+     * This is also needed to ensure the driver gets initialized properly.
+     */
+    if (v4l2_ioctl (v4l2object->video_fd, VIDIOC_S_INPUT, &input) >= 0) {
+        if (v4l2_ioctl (v4l2object->video_fd, VIDIOC_G_INPUT, &input) < 0) {
+            GST_ELEMENT_ERROR (e, RESOURCE, SETTINGS,
+                (_("Failed to get input %d in device '%s'."),
+                    n, v4l2object->videodev), GST_ERROR_SYSTEM);
+            return FALSE;
+        }
+    }
+    #endif
+
     GST_LOG_OBJECT (e, "   index:     %d", input.index);
     GST_LOG_OBJECT (e, "   name:      '%s'", input.name);
     GST_LOG_OBJECT (e, "   type:      %08x", input.type);
-- 
1.6.3.3

